### PR TITLE
COMMERCE-1187 paypal add discounts and shipping as order items

### PR DESCRIPTION
**Pull request by luca-pellizzon**
@marco-leo

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

File: commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java [hunk 1]
```diff
@@ -847,16 +837,73 @@ private ItemList _getItemList(
...
+	private String _getLabel(Locale locale, String key) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(resourceBundle, key);
+	}
+
+	private void _getOrderDiscounts(
+		CommerceOrder commerceOrder, CommerceCurrency commerceCurrency,
+		Locale locale, List<Item> items) {
```
Comment (marco-leo) [inline commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java]: don't use references object as paraemter. Instead return a new list of discounted items

File: commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java [hunk 2]
```diff
@@ -847,16 +837,73 @@ private ItemList _getItemList(
...
+
+		BigDecimal subtotalDiscountAmount =
+			commerceOrder.getSubtotalDiscountAmount();
+
+		if ((subtotalDiscountAmount != null) &&
+			(subtotalDiscountAmount.compareTo(BigDecimal.ZERO) > 0)) {
+
+			Item item = new Item();
```
Comment (marco-leo) [inline commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java]: encapsulate in a new method

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by marco-leo** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Pull request merged**